### ISSUE TITLE
Getting error open GHOpenPR on litee/gh.nvim latest

### ISSUE THREAD

**Issue by marcomayer** [ldelossa/gh.nvim#48]
I'm getting this error and am not sure what might cause it. I use the default setup() and tried to set the icon_set options to nil or "" but that didn't help:

```
Error executing vim.schedule lua callback: ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:66: attempt to index upvalue 'icon_set'
 (a nil value)
stack traceback:
        ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:66: in function 'marshal_node'
        ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:153: in function '_marshal_tree'
        ...pack/packer/start/litee.nvim/lua/litee/lib/tree/init.lua:306: in function 'write_tree'
        ...pack/packer/start/litee.nvim/lua/litee/lib/tree/init.lua:315: in function 'write_tree_no_guide_leaf'
        ...vim/site/pack/packer/start/gh.nvim/lua/litee/gh/init.lua:53: in function 'pre'
        ...ack/packer/start/litee.nvim/lua/litee/lib/panel/init.lua:129: in function 'toggle_panel'
        ...e/pack/packer/start/gh.nvim/lua/litee/gh/pr/handlers.lua:154: in function 'ui_handler'
        ...e/pack/packer/start/gh.nvim/lua/litee/gh/pr/handlers.lua:179: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by ldelossa** [comment]
@marcomayer if you specify "default" does it work?

**Comment by marcomayer** [comment]
> @marcomayer if you specify "default" does it work?

yes now it does and oddly I can't reproduce it anymore. Sorry, probably was some local config-cache issue!

**Comment by ldelossa** [comment]
Cool :)

**Comment by ofseed** [comment]
Some issue... but using `icon_set = "nerd"` could not resolve this problem...
```
⣾ successfully fetched all PR state.
Error executing vim.schedule lua callback: ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:66: attempt to in
dex upvalue 'icon_set' (a nil value)
stack traceback:
        ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:66: in function 'marshal_node'
        ...ker/start/litee.nvim/lua/litee/lib/tree/marshal/init.lua:153: in function '_marshal_tree'
        ...pack/packer/start/litee.nvim/lua/litee/lib/tree/init.lua:306: in function 'write_tree'
        ...pack/packer/start/litee.nvim/lua/litee/lib/tree/init.lua:315: in function 'write_tree_no_guide_leaf'
        .../nvim/site/pack/packer/opt/gh.nvim/lua/litee/gh/init.lua:53: in function 'pre'
        ...ack/packer/start/litee.nvim/lua/litee/lib/panel/init.lua:129: in function 'toggle_panel'
        ...ite/pack/packer/opt/gh.nvim/lua/litee/gh/pr/handlers.lua:154: in function 'ui_handler'
        ...ite/pack/packer/opt/gh.nvim/lua/litee/gh/pr/handlers.lua:179: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by ldelossa** [comment]
@ofseed you have the latest version of both "litee.nvim" and "gh.nvim" right?

**Comment by ofseed** [comment]
`litee.nvim` and `gh.nvim` are what I just installed today, and the error occurs when I run `GHOpenPR` for the first time

**Comment by ldelossa** [comment]
@ofseed can you show me your config for both litee and gh.nvim? Ill try to reproduce

**Comment by ofseed** [comment]
I separate the configuration file of every plugin into an individual file. This is [the configuration file of gh.nvim](https://github.com/ofseed/nvim/blob/main/lua/tool/gh.lua), and [this is how I call it](https://github.com/ofseed/nvim/blob/447df0a51dde753a31075c653944fd2800c7ca8d/lua/plugins.lua#L640). As you can see I copied almost the entire default configurationin the `README.md`.

**Comment by ldelossa** [comment]
@ofseed hey there, I just reproduced your config. I can't seem to reproduce the issue on my end. 

I changed my config to this:

```
require('litee.lib').setup()
require('litee.gh').setup({
    icon_set = "nerd",
})
```

and opened a PR, nerd icons rendered fine with no error splat. 

If you are using Packer, it does some module caching, maybe something is odd with this

**Comment by marcomayer** [comment]
@ofseed For me, it was probably a cache issue, I did PackerCompile a couple of times, reinstalled the packages, and then the issue was gone. Also might want to try to set 
```require("litee.lib").setup({icon_set = "nerd"}```
explicitly.

**Comment by dlvhdr** [comment]
I tried everything in this thread but nothing worked.
For some reason, `litee.lib` is not set up when `gh.nvim` is loaded and so `icon_set` is not set and is `nil`.
Couldn't figure out why, maybe it's a but with packer.

Anyway, what I did was load the config lazily by doing this:

```lua
  use({
    "ldelossa/gh.nvim",
    cmd = { "GHOpenPR" },
    config = function()
      require("dlvhdr.gh-nvim")
    end,
    requires = { { "ldelossa/litee.nvim" } },
  })
```

I also had to remove the `pcall` calls and do a straight up `require`.
With these 2 things - it works.
